### PR TITLE
feat(llama-index): add embeddings

### PR DESCRIPTION
**Pull request by David-Kristek**
Refs: #311

Add embedding support for llamaindex.

### CODE REVIEW

File: examples/extensions/llama_index/llama_index_embedding.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""
+Llamaindex Embeddings
```
Comment (Tomas2D) [inline examples/extensions/llama_index/llama_index_embedding.py:2]: It's `LlamaIndex`

File: src/genai/extensions/llama_index/__init__.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 """Extension for LLamaIndex library"""
 
+from genai.extensions.llama_index.embeddings import LlamaIndexEmbeddingsInterface
 from genai.extensions.llama_index.llm import IBMGenAILlamaIndex
 
-__all__ = ["IBMGenAILlamaIndex"]
+__all__ = ["IBMGenAILlamaIndex", "LlamaIndexEmbeddingsInterface"]
```
Comment (Tomas2D) [inline src/genai/extensions/llama_index/__init__.py:6]: Be consistent with naming. -> `IBMGenaAILlamaIndexEmbedding`

File: src/genai/extensions/llama_index/embeddings.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        return await asyncio.get_running_loop().run_in_executor(None, self._get_text_embeddings, texts)
+
+    def _get_embeddings(self, texts: list[str]) -> list[Embedding]:
+        embeddings: list[list[float]] = []
+        for response in self.client.text.embedding.create(
+            model_id=self.model_id, inputs=texts, parameters=self.parameters, execution_options=self.execution_options
+        ):
+            embeddings.extend(response.results)
+
+        return embeddings
```
Comment (Tomas2D) [inline src/genai/extensions/llama_index/embeddings.py:53]: There is no `_aget_embeddings`?

File: tests/integration/extensions/cassettes/test_llama_index_embeddings/TestLangChainEmbeddings.test_async_embedding.yaml
Comment (Tomas2D) [inline tests/integration/extensions/cassettes/test_llama_index_embeddings/TestLangChainEmbeddings.test_async_embedding.yaml:1]: Wrong file names.

File: tests/integration/extensions/cassettes/test_llama_index_embeddings/TestLangChainEmbeddings.test_embedding.yaml
Comment (Tomas2D) [inline tests/integration/extensions/cassettes/test_llama_index_embeddings/TestLangChainEmbeddings.test_embedding.yaml:1]: Wrong file name.

File: tests/integration/extensions/test_llama_index_embeddings.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import pytest
+
+from genai.extensions.llama_index import LlamaIndexEmbeddingsInterface
+from genai.schema import TextEmbeddingCreateEndpoint, TextEmbeddingParameters
+
+
+@pytest.mark.integration
+class TestLangChainEmbeddings:
```
Comment (Tomas2D) [inline tests/integration/extensions/test_llama_index_embeddings.py:8]: wrong name

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: src/genai/extensions/llama_index/embeddings.py (hunk 3 above)
Comment (David-Kristek) [reply src/genai/extensions/llama_index/embeddings.py:53]: this method is: `_aget_text_embeddings`

**Review by Tomas2D — APPROVED** [review]
👍🏻 🚀

**Pull request merged**